modules/related-posts/jetpack-related-posts.php

```diff
@@ -175,12 +177,37 @@ public function filter_add_target_to_dom( $content ) {
 		}
 
 		if ( ! $this->_found_shortcode ) {
-			$content .= "\n" . $this->get_target_html();
+			if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+				$content .= "\n" . $this->get_server_rendered_html();
+			} else {
+				$content .= "\n" . $this->get_client_rendered_html();
+			}
 		}
 
 		return $content;
 	}
 
+	/**
+	 * Render static markup based on the Gutenberg block code
+	 *
+	 * @return string Rendered related posts HTML.
+	 */
+	public function get_server_rendered_html() {
+		$rp_settings       = Jetpack_Options::get_option( 'relatedposts', array() );
```

Comment:
@gravityrail I think this should have been `$this->get_options()` instead of `Jetpack_Options::get_option( 'relatedposts', array() )`. See #14386.